Assets/MirageXR/ContentTypes/3DModel/Model.cs

```diff
@@ -126,65 +133,92 @@ private async Task<bool> LoadGltf(ToggleObject content)
                 return false;
             }
 
-            OnFinishLoading(transform.Find("Sketchfab_model").gameObject, _gltf.GetAnimationClips());
+            var legacyAnimation = instantiator.SceneInstance.LegacyAnimation;
+            if (legacyAnimation != null)
+            {
+                Debug.Log("playing animation (with data from gltfast scene instantiator).");
+
+                legacyAnimation.clip = _gltf.GetAnimationClips()[0];
+                legacyAnimation.clip.wrapMode = UnityEngine.WrapMode.Loop;
+                legacyAnimation.clip.legacy = true;
+                legacyAnimation.clip.EnsureQuaternionContinuity();
+                legacyAnimation.playAutomatically = true;
+                legacyAnimation.Play(_gltf.GetAnimationClips()[0].name);
+            }
+
+            OnFinishLoading(transform.Find("Sketchfab_model").gameObject); // , _gltf.GetAnimationClips()
             return true;
         }
 
-        private void OnFinishLoading(GameObject model, AnimationClip[] clip)
+        private void OnFinishLoading(GameObject model) // , AnimationClip[] clip
         {
             if (this == null)
             {
                 Destroy(model);
                 return;
             }
 
+            
             var modelTransform = transform;
             var startPos = modelTransform.position + modelTransform.forward * -0.5f + modelTransform.up * -0.1f;
-
+            
             model.transform.SetParent(modelTransform);
 
+            /*
+             * ALEX: please check this - I don't think it is needed and it prevented the animation 
+             * from being played
+             * 
```

Comment:
Yes, I don't think it's necessary. but we need to understand that models can have different sizes and it would be nice to have a tool that will fit the loaded model into an adequate size (also so that we can specify the original size of the model)

Reply:
I am not quite sure what broke it - there was a warning floating by that said something about "size set to infinite" or some such, so I guess it was simply too small for the animation component to accept.

Reply:
I checked again, and it turns out the mode renaming was the real culprit, so I restored all the resize functions and only commented this out:
```
// model.name = _obj.option;
```
Is this model.name relevant anywhere? I think not, right?

Reply:
I checked carefully - deleting, adding, keep alive, upload/download, all works. Merging.